lib/pmdtester/parsers/pmd_report_document.rb

```diff
@@ -28,11 +29,11 @@ def start_element(name, attrs = [])
       case name
       when 'file'
         @current_violations = []
-        @current_filename = attrs['name']
+        @current_filename = attrs['name'].sub(/^#{@working_dir}/, '')
       when 'violation'
         @current_violation = PmdViolation.new(attrs, @branch_name)
       when 'error'
-        @current_filename = attrs['filename']
+        @current_filename = attrs['filename'].sub(/^#{@working_dir}/, '')
```

Comment:
That's again only the hash key. `attrs['filename']` is not changed, so the full filename is later on still available in PmdError.